bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/ColorValue.java

```diff
@@ -33,46 +37,44 @@
  * Accepts user updates from a HSBType, OnOffType and StringType.
  * </p>
  * Accepts MQTT state updates as OnOffType and a
- * StringType with comma separated HSB ("h,s,b"), RGB ("r,g,b") and on, off strings.
+ * StringType with comma separated HSB ("h,s,b"), RGB ("r,g,b"), CIE xyY ("x,y,Y") and on, off strings.
  * On, Off strings can be customized.
  *
  * @author David Graeff - Initial contribution
+ * @author Aitor Iturrioz - Add CIE xyY colors support
  */
 @NonNullByDefault
 public class ColorValue extends Value {
-    private final boolean isRGB;
+    private final Logger logger = LoggerFactory.getLogger(ColorValue.class);
+    private final String colorMode;
     private final String onValue;
     private final String offValue;
     private final int onBrightness;
 
     /**
      * Creates a non initialized color value.
      *
-     * @param isRGB True if this is an RGB color value instead of a HSB one.
+     * @param colorMode The color mode: "hsb", "rgb" or "xyY".
      * @param onValue The ON value string. This will be compared to MQTT messages.
      * @param offValue The OFF value string. This will be compared to MQTT messages.
      * @param onBrightness When receiving a ON command, the brightness percentage is set to this value
      */
-    public ColorValue(boolean isRGB, @Nullable String onValue, @Nullable String offValue, int onBrightness) {
+    public ColorValue(String colorMode, @Nullable String onValue, @Nullable String offValue, int onBrightness) {
```

Comment:
I would prefere to use an enum for the colorMode here.

Reply:
Agreed.

Reply:
Done!